httpcomm/communicatewithjson.go

```diff
@@ -0,0 +1,137 @@
+package httpcomm
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+
+	"github.com/arquivei/foundationkit/errors"
+	"github.com/arquivei/foundationkit/trace"
+)
+
+// CommunicateWithJSON uses a given @httpClient to communicate with a HTTP
+// server at @fullURL using the specified @httpMethod. This function will
+// send @requestData marshalled as a JSON content and wait for the server
+// response. The response will be written into the given @outResponse (it
+// will mutate the parameter contents) using the json unmarshaling rules set
+// by @outResponse itself.
+//
+// To reduce damage in case of bugs or attacks, a @maxAcceptedBodySize must be
+// passed, and bodies contents larger than this will be considered noxious and
+// return an error.
+//
+// A value of @maxErrBodySize must be passed to indicate how much of response
+// contents can be added into the error message.
+func CommunicateWithJSON(
+	ctx context.Context,
+	httpClient http.Client,
+	httpMethod string,
+	fullURL string,
+	requestData interface{},
+	maxAcceptedBodySize int64,
```

Comment:
This seems to be a rather argument to think when using this function. Could you wrap  maxAcceptedBodySize and maxErrBodySize into a config structure? Then I could use a "defaultHttpConfig" in my code and make everything easier.

Reply:
It's possible to wrap both `maxAcceptedBodySize` and `maxErrBodySize` into a structure as much as it is possible to also wrap other parameters along the way. It might be a case of using a higher level function or structure with better abstractions built on top of this function. Didn't work on it as there is still too few uses to measure what could be most useful. In this changeset, I'll leave it as-is.